NetworkConfigurations/NetworkAddressingAndFilteringSubCategories/DHCP.md

```diff
@@ -0,0 +1,263 @@
+
+```
+sudo systemctl restart isc-dhcp-server
+Job for isc-dhcp-server.service failed because the control process exited with error code.
+See "systemctl status isc-dhcp-server.service" and "journalctl -xe" for details.
+```
+
+This issue arises because the IP address assigned to the interface of DENT needs to be appropriate for DHCP server usage. For instance, if a bridge interface named br0 exists, it must have an IP address from which other devices can be assigned IPs within the DHCP server range.
+
+To assign an IP address to br0, execute the following command:
+
+```
+ip address add 192.168.1.10/24 dev br0
+```
+
+Afterward, restart the DHCP server using `sudo systemctl restart isc-dhcp-server`. This time, it should operate without errors.
+
+Additionally, verify the status to ensure it's active and running:
+
+```
+systemctl status isc-dhcp-server.service
+● isc-dhcp-server.service - LSB: DHCP server
+   Loaded: loaded (/etc/init.d/isc-dhcp-server; generated; vendor preset: enable
+   Active: active (running)
+```
+
+Now, the DHCP server is ready to randomly assign IP addresses to different devices from the specified range.
+
+### Example:
+
+**Using DHCP with a Bridge:**
+
+1. Begin by creating a bridge interface.
+
+   ```
+   ip link add name br0 type bridge
+   ```
+
+2. Set the desired ports (`enp0s4` and `enp0s5`) as members of the bridge.
+
+   ```
+   ip link set enp0s4 master br0
+   ip link set enp0s5 master br0
+   ```
+
+3. Bring the interface 'br0' and the ports `enp0s4` and `enp0s5` up.
+
+   ```
+   ip link set br0 up
+   ip link set enp0s4 up
+   ip link set enp0s5 up
+   ```
+
+4. Initiate the DHCP client on the bridge interface (`br0`).
+   ```
+   dhclient br0
+   ```
+
+Please ensure that you add an IP address to the `br0` interface so that it follows the range of the DHCP server.
+
+**Using DHCP with a VLAN Sub-interface:**
+
+1. Establish a bridge interface with VLAN filtering enabled.
+
+   ```
+   ip link add name br0 type bridge vlan_filtering 1
+   ```
+
+2. Bring the bridge interface `br0` up.
+
+   ```
+   ip link set dev br0 up
+   ```
+
+3. Set the desired ports (`enp0s4` and `enp0s5`) as members of the bridge.
+
+   ```
+   ip link set dev enp0s4 master br0
+   ip link set dev enp0s5 master br0
+   ```
+
+4. Add the VLAN membership for the desired VLAN (e.g., VLAN ID 10) on the ports `enp0s4` and `enp0s5` and the bridge interface `br0`.
+
+   ```
+   bridge vlan add vid 10 dev enp0s4
+   bridge vlan add vid 10 dev enp0s5
+   bridge vlan add vid 10 dev br0 self
+   ```
+
+5. Generate VLAN sub-interfaces (`br0.10` and `br0.20` for example) associated with the respective VLAN IDs (e.g., VLAN ID 10 and VLAN ID 20).
+
+   ```
+   ip link add link br0 name br0.10 type vlan id 10
+   ip link add link br0 name br0.20 type vlan id 20
+   ```
+
+6. Initiate the DHCP client on the VLAN sub-interfaces (`br0.10` and `br0.20`).
+
+   ```
+   dhclient br0.10
+   dhclient br0.20
+   ```
```

Comment:
can you please elaborate on why you are using the local dhcp server to assign IP addresses for local interfaces.

Reply:
Considering the various options for configuring a DHCP client using bridge and VLAN interfaces as outlined [here](https://github.com/Marvell-switching/switchdev-prestera/wiki/DHCP), I have included this information. Although another option involving the use of a bond was available, it appears that the DENT VM does not support bonding, although it does function on hardware. In any case, we can proceed by having the information for the bridge interface and removing the VLAN configuration here.